[Docs] Layer content update (#5843)
grommet/hpe-design-system: apps/docs/src/pages/components/layer.mdx

import { Button, Notification } from 'grommet';
import {
  AccessibilitySection,
  BestPracticeGroup,
  Example,
  StyleTable,
} from '../../layouts';
import {
  ActionLabels,
  ActionLabelTitle,
  CenterInformational,
  ContentDrivenLayout,
  DoubleConfirmationAnatomy,
  DoubleConfirmationBestPractice,
  ConfigurationForm,
  FilteringTable,
  LayerAnatomy,
  LayerExample,
  LayerCenterExample,
  LayerNotificationExample,
  LayerSideDrawerExample,
  LayerStickyExample,
  LayerStickyScrollExample,
  ActionableLayerClose,
  InformationalLayerClose,
  LayerTypes,
} from '../../examples';
import { Checkmark } from '@hpe-design/icons-grommet';

<Example
  componentName="Layer"
  docs="https://v2.grommet.io/layer?theme=hpe#props"
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/components/layer/LayerExample.js"
  figma="https://www.figma.com/design/HDckqS2MWhINfC8EIQPMV1/HPE-Design-System-Components-V2?m=auto&node-id=2567-534&t=6zfWJlYNDdZ4YwHx-1"
  grommetSource="https://github.com/grommet/grommet/blob/master/src/js/components/Layer/Layer.js"
>
  <LayerExample />
</Example>

## Use cases

### Confirming next steps

> Use center layer when a process has been successfully initiated and the user needs a clear, unobtrusive confirmation. It is suitable for situations where no further action is required, but the user should be informed of what is happening next—for example, confirming that a firmware update has started and providing a brief expectation of what will follow.

Reviewer: ```suggestion
Use [center layer](#center) when a process has been successfully initiated and the user needs a clear, unobtrusive confirmation. It is suitable for situations where no further action is required, but the user should be informed of what is happening next—for example, confirming that a firmware update has started and providing a brief expectation of what will follow.
```

<Example
  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/components/layer/CenterInformational.js"
  github="https://github.com/grommet/hpe-design-system/blob/master/apps/docs/src/examples/components/layer/CenterInformational.js"
  showResponsiveControls={['fullScreen']}
  screenContainer
  height="650px"
  width="100%"
>
  <CenterInformational />
</Example>

### Confirming destructive deletion

Highly destructive actions refer to use cases where large amounts of data will be deleted
and cannot be restored. In these instances, have the user type a string that matches the
name or path of the object to be deleted. This approach ensures the user is aware of the

> consequences of the action.

Reviewer: @julauxen I think we might be too prescriptive here. Can we try something more general?

```suggestion
and cannot be restored. In these instances, have the user confirm their intent by entering a specific value in an input. This approach ensures the user acknowledges the consequences of the action.
```

<Example
  code={[
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/content-layouts/ContentDrivenLayout.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-core/src/js/components/layouts/ModalDialog/ModalDialog.js',
  ]}
  github="https://github.com/grommet/hpe-design-system/blob/master/apps/docs/src/examples/templates/content-layouts/ContentDrivenLayout.js"
  showResponsiveControls={['fullScreen']}
  screenContainer
  height="650px"
  width="100%"
>
  <ContentDrivenLayout />
</Example>

### Adding a new record

> Use side drawer layer when adding a new item or configuring details without 

Reviewer: ```suggestion
Use [side drawer layer](#side-drawer) when adding a new item or configuring details without 
```

navigating away from the main view. In this example, selecting Add application 
opens a side drawer containing a form, allowing the user to input configuration 
details while still referencing the existing list of applications.

<Example
  docs="https://v2.grommet.io/layer?theme=hpe#props"
  code={[
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/components/layer/ConfigurationForm.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-core/src/js/components/layouts/Layer/LayerHeader.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/components/layer/components/ConfirmationContext.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/components/layer/components/DoubleConfirmation.js',
  ]}
  github="https://github.com/grommet/hpe-design-system/blob/master/apps/docs/src/examples/components/layer/ConfigurationForm.js"
  height={{ min: 'small' }}
  template
  showResponsiveControls={[]}
>
  <ConfigurationForm />
</Example>

### Filtering data

> Use side drawer layer when a list contains many items and users need tools to quickly 

Reviewer: ```suggestion
Use [side drawer layer](#side-drawer) when a list contains many items and users need tools to quickly 
```

    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/FilterControls/Filters.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/FilterControls/FiltersLayer.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/FilterControls/ResultsSummary.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/FilterControls/SearchFilter.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/FilterControls/filterTypes/FilterCheckBoxGroup.js',
    'https://raw.githubusercontent.com/grommet/hpe-design-system/master/apps/docs/src/examples/templates/filtering/FilteringTable/mockData.js',
  ]}
  github="https://github.com/grommet/hpe-design-system/blob/master/apps/docs/src/examples/templates/filtering/FilteringTable/FilteringTable.js"
  template
  showResponsiveControls={[]}
>
  <FilteringTable />
</Example>

### Completing a multi-step process

> Use full screen layers when a task needs full user’s attention. See [Wizard](/templates/wizard) for more detail.

Reviewer: ```suggestion
Use [full screen layers](#fullscreen) when a task needs full user’s attention. See [Wizard](/templates/wizard) for more detail.
```

## Anatomy

<Example caption="Diagram illustrating the parts composing Layer." plain>
  <LayerAnatomy />
</Example>

| Label  | Region                | Purpose                                                                                 |                  Required                   | Notes                                                                                                                                                       |
| ------ | --------------------- | --------------------------------------------------------------------------------------- | :-----------------------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | **1**  | **Header**            | Identifies the layer.                                                                   | <Checkmark a11yTitle="true" size="small" /> | --                                                                                                                                                          |

Reviewer: centers checkmark

```suggestion
| **1**  | **Header**            | Identifies the layer.                                                                   | <span><Checkmark a11yTitle="true" size="small" /></span> | --                                                                                                                                                          |
```

> | **1a** | **Title**             | A short, descriptive title identifying the layer.                                       | <Checkmark a11yTitle="true" size="small" /> | <span>Use H2 for layer titles unless wrapped in a semantic section that permits H1. See [accessibility](#accessibility).</span>                  |

Reviewer: ```suggestion
| **1a** | **Title**             | A short, descriptive title identifying the layer.                                       | <span><Checkmark a11yTitle="true" size="small" /></span> | <span>Use H2 for layer titles unless wrapped in a semantic `<section>` element that permits H1. See [accessibility](#accessibility).</span>                  |
```

require major context switching. See [Confirming next steps](#confirming-next-steps) and [Confirming destructive deletion](#confirming-destructive-deletion).

### Side drawer
Side drawer layers are designed to let users view details, configure items, or adjust filters 
while still maintaining full visibility of their main page. See [Adding a new record](#adding-a-new-record) and 
[Filtering data](#filtering-data).

Apply these [layer properties](https://v2.grommet.io/layer?theme=hpe#props) for a side drawer layer.

| Property                                         | Value    | Notes                                                                                               |
| ------------------------------------------------ | -------- | --------------------------------------------------------------------------------------------------- |
| [position](https://v2.grommet.io/layer#position) | right    | --                                                                                                  |
| [full](https://v2.grommet.io/layer#full)         | vertical | --                                                                                                  |

> | [modal](https://v2.grommet.io/layer#modal)       | false    | A non-modal layer allows the underlying page to remain interactive while the layer is open.         |

Reviewer: @julauxen Should we provide both true and false as options? For example, the "Filtering data" example is executed with `modal="true"`.

```suggestion
| [modal](https://v2.grommet.io/layer#modal)       | false<br />true    | <span>`false` provides a non-modal layer allowing the underlying page to remain interactive while the layer is open.</span><br /><span>`true` provides a modal layer that blocks interaction with the underlying page while the layer is open.</span>         |
```

Reviewer 2: can't seem to apply this suggestion. @Sulaymon333 could you help here?

Author: @julauxen this is applied already via this [commit](https://github.com/grommet/hpe-design-system/pull/5843/changes/c3adb445385ff41fa352153e19622959becbe1b3).

### Fullscreen
Fullscreen layers support more extensive or content‑heavy tasks that need additional space and 
focus. See [Completing a multi-step process](#completing-a-multi-step-process).

Apply these [layer properties](https://v2.grommet.io/layer?theme=hpe#props) for a fullscreen layer.

| Property                                         | Value    |
| ------------------------------------------------ | -------- |
| [full](https://v2.grommet.io/layer#full)         | true |

## Dos and do nots 

<BestPracticeGroup>
  <Example
    height={{ min: 'xsmall' }}
    bestPractice={{
      type: 'do',
      message: `Match the primary action verb to that of the layer's title.`,
    }}
    width="100%"
  >

>     <ActionLabelTitle />

Reviewer: Please make sure that the buttons are right-aligned (their order will reverse when right-aligned as well).

See [previous comment](https://www.figma.com/design/gkAlhfnBI1OrqAB2sVHObp?node-id=831-4229#1621341803).

cc: @julauxen

Reviewer 2: sorry, I missed that during the figma stage. @Sulaymon333 feel free to make the change

Author: @halocline this is addressed via this [commit](https://github.com/grommet/hpe-design-system/pull/5843/changes/885f407049d25fcbfbe51a3084421c9b9a61bf63).
